docs/technical/preservation/file_repair.md

```diff
@@ -0,0 +1,72 @@
+# How to Demo File Repair
+
+## Create resources
+In the figgy staging instance, create 3 scanned resources which each have one image. Name them clearly so you can tell which is which, e.g.
+  * Test Repair Bad Cloud File
+  * Test Repair Bad Local File
+  * Test Repair Bad Cloud and Local Files
+
+Complete each resource so they all preserve to the cloud.
+
+## Corrupt a local file
+You'll want to corrupt the local file for both the "bad local file" and the "bad
+cloud and local files" objects.
+
+`$ bundle exec cap staging rails:console`
+
+```ruby
+# set an id variable to your fileset's id
+qs = ChangeSetPersister.default.query_service
+fs = qs.find_by(id: id)
+fs.file_metadata.first.file_identifiers.first
+```
+
+This will print the disk location of the local file. ssh to the staging box and
+cd to that location. then you can corrupt it by appending something to the end
+of it, e.g.
+```
+% echo "p0wn3d" >> [filename].tif
+```
+
+## Corrupt a cloud file
+You'll want to corrupt the cloud file for both the "bad cloud file" and the "bad
+cloud and local files" objects.
+
+`$ bundle exec cap staging rails:console`
+
+```
+# set an id variable to your fileset's id
+qs = ChangeSetPersister.default.query_service
+fs = qs.find_by(id: id)
+po = Wayfinder.for(fs).preservation_object
+po.binary_nodes.first.file_identifiers.first.id
```

Comment:
Similar, can we override this from the rails console somehow? Maybe...

```ruby
original_file = Valkyrie::StorageAdapter.find_by(id: po.binary_nodes.first.file_identifiers.first.id)
storage_adapter = Valkyrie::StorageAdapter.find(:google_cloud_storage)
storage_adapter.upload(file: File.open(Rails.root.join("spec", "fixtures", "files", "bad.tif")), original_filename: po.binary_nodes.first.original_filename.first, resource: fs)
```

Untested, but worth a shot?